src/main/java/hydroguard/api/mock/controllers/friends/FriendsController.java

```diff
@@ -0,0 +1,95 @@
+package hydroguard.api.mock.controllers.friends;
+
+import hydroguard.api.mock.models.friends.AddFriendDTO;
+import hydroguard.api.mock.models.friends.Friends;
+import hydroguard.api.mock.models.friends.UpdateFriendStatusDTO;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/friends")
+public class FriendsController {
+
+    private final List<Friends> friendsList = new ArrayList<>();
+
+    @PostMapping("/invite")
+    public ResponseEntity<String> inviteFriend(@RequestBody AddFriendDTO dto) {
+        Friends newFriendship = new Friends(dto.getUserId1(), dto.getUserId2());
+        friendsList.add(newFriendship);
+
+        return ResponseEntity.ok("Friend invitation sent from " + dto.getUserId1() + " to " + dto.getUserId2());
+    }
+
+    @PostMapping("/invite/accept")
+    public ResponseEntity<String> acceptInvitation(@RequestBody AddFriendDTO dto) {
+        for (Friends friendship : friendsList) {
+            if (friendship.getUserId1().equals(UUID.fromString(dto.getUserId1())) &&
+                    friendship.getUserId2().equals(UUID.fromString(dto.getUserId2())) &&
+                    friendship.getStatus() == Friends.Status.PENDING) {
+
+                friendship.setStatus(Friends.Status.ACCEPTED);
+                return ResponseEntity.ok("Friendship accepted between " + dto.getUserId1() + " and " + dto.getUserId2());
+            }
+            if (friendship.getUserId2().equals(UUID.fromString(dto.getUserId1())) &&
+                    friendship.getUserId1().equals(UUID.fromString(dto.getUserId2())) &&
+                    friendship.getStatus() == Friends.Status.PENDING) {
+
+                friendship.setStatus(Friends.Status.ACCEPTED);
+                return ResponseEntity.ok("Friendship accepted between " + dto.getUserId1() + " and " + dto.getUserId2());
+            }
+        }
+        return ResponseEntity.status(404).body("Friend invitation not found.");
+    }
+
+    @PutMapping("/{id}")
```

Comment:
Waar is deze method voor nodig? Kan je niet beter een `/invite/deny` maken?

Reply:
Voor mogelijke andere friendship dingen die later geïmplementeerd kunnen worden